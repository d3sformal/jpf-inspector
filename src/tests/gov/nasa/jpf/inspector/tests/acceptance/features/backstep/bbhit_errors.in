# All bbhit's in this file cause an error.

# Not yet running:
back_breakpoint_hit && wait

cr bp local_write = a
run && wait

# Not yet in transition:
back_breakpoint_hit && wait

step_over && wait

# Multistep not permitted:
back_breakpoint_hit 1 && wait

# Incorrect abbreviation:
back_bbhit && wait

cr bp local_write = c
run && wait
bso && wait
# Instruction in the future:
bbhit && wait
bso && wait
bso && wait

# Transition in the future:
bbhit && wait