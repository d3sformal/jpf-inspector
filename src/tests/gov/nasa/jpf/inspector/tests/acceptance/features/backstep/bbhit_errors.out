ERR: This command may only be used when JPF has started and is paused. If JPF is not already started, then start it using 'run' and if it is already running, then pause it using 'break'. Then you will be able to execute this command. Alternatively, you may disable safe mode in the configuration file by setting 'jpf-inspector.safe-mode = false'.
INFO: JPF created and connected, SuT is started
INFO: Breakpoint hit: [1 :  state=ENABLED hits=1 hitsTotal=1 local_write=a]
ERR: You have already backtracked through the instruction where the last breakpoint was hit (target instruction not found).
INFO: SuT is stopped
	SuT (thread 0) will now execute gov/nasa/jpf/inspector/tests/acceptance/sut/MultiTransitionSinglePath.java:9 (iconst_1), source: int transitionStart = Verify.getInt(1, 2);
ERR: Parse error: extraneous input '1' expecting {<EOF>, '&&', WS}
back_breakpoint_hit 1 && wait
                    ^
ERR: 'back_bbhit' is not recognized as an internal command, alias or a custom command.
INFO: Breakpoint hit: [3 :  state=ENABLED hits=1 hitsTotal=1 local_write=c]
INFO: SuT is stopped
	SuT (thread 0) will now execute gov/nasa/jpf/inspector/tests/acceptance/sut/MultiTransitionSinglePath.java:10 (iload_1), source: int b = a + 4;
ERR: You have already backtracked through the instruction where the last breakpoint was hit (target instruction not found).
INFO: SuT is stopped
	SuT (thread 0) will now execute gov/nasa/jpf/inspector/tests/acceptance/sut/MultiTransitionSinglePath.java:9 (iconst_1), source: int transitionStart = Verify.getInt(1, 2);
INFO: SuT is stopped
	SuT (thread 0) will now execute gov/nasa/jpf/inspector/tests/acceptance/sut/MultiTransitionSinglePath.java:8 (iconst_5), source: int a = 5;
ERR: You have already backtracked through the transition where the last breakpoint was hit (not enough transitions).
INFO: Breakpoint hit: [1 :  state=ENABLED hits=2 hitsTotal=2 local_write=a]
INFO: JPF Terminating